docs/connector-development/connector-specification-reference.md

```diff
@@ -96,12 +96,14 @@ In this case, use the `"airbyte_hidden": true` keyword to hide that field from t
 }
 ```
 
+Results in the following form:
+
 ![hidden fields](../.gitbook/assets/spec_reference_hidden_field_screenshot.png)
 
-Results in the following form: 
 
+## Airbyte Modifications to `jsonschema`
```

Comment:
Making this more explicit means we can hopefully start enforcing other things if we want to. For example pertaining to enum values, we could also enforce:
* if the `default` keyword is used, there is an item called "default", and the default item is also in the list
* that all options are of the type listed

(Ideas from here: https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/destination-kafka/src/main/resources/spec.json#L188-L195)

These restrictions AFAIK come from the frontend, so If we wanted to move forward with enforcing more things for better UX up front, I'd say we should be in contact with the frontend team to align on a big list of things to enforce so that it could all be done at once and they know explicitly what they want to enforce, in case that changes.